zrml/prediction-markets/src/migrations.rs

```diff
@@ -15,6 +15,408 @@
+
+        let new_markets = storage_iter::<OldMarketOf<T>>(MARKET_COMMONS, MARKETS)
+            .map(|(key, old_market)| {
+                total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+                let advisory = if old_market.creation == MarketCreation::Advised {
+                    Some(Bond {
+                        who: old_market.creator.clone(),
+                        value: T::AdvisoryBond::get(),
+                        is_settled: old_market.status != MarketStatus::Proposed,
+                    })
+                } else {
+                    None
+                };
+                let oracle = Some(Bond {
+                    who: old_market.creator.clone(),
+                    value: T::OracleBond::get(),
+                    is_settled: old_market.status == MarketStatus::Resolved,
+                });
+                let validity = if old_market.creation == MarketCreation::Permissionless {
+                    Some(Bond {
+                        who: old_market.creator.clone(),
+                        value: T::ValidityBond::get(),
+                        is_settled: old_market.status == MarketStatus::Resolved,
+                    })
+                } else {
+                    None
+                };
+                let new_market = Market {
+                    creator: old_market.creator,
+                    creation: old_market.creation,
+                    creator_fee: old_market.creator_fee,
+                    oracle: old_market.oracle,
+                    metadata: old_market.metadata,
+                    market_type: old_market.market_type,
+                    period: old_market.period,
+                    scoring_rule: old_market.scoring_rule,
+                    status: old_market.status,
+                    report: old_market.report,
+                    resolved_outcome: old_market.resolved_outcome,
+                    dispute_mechanism: old_market.dispute_mechanism,
+                    deadlines: old_market.deadlines,
+                    bonds: MarketBonds { advisory, oracle, validity },
+                };
+                (key, new_market)
+            })
+            .collect::<Vec<_>>();
+
+        for (key, new_market) in new_markets {
+            put_storage_value::<MarketOf<T>>(MARKET_COMMONS, MARKETS, &key, new_market);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+        }
+
+        StorageVersion::new(MARKET_COMMONS_NEXT_STORAGE_VERSION).put::<MarketCommonsPallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        log::info!("RecordBonds: Done!");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        let old_markets =
+            storage_iter::<OldMarketOf<T>>(MARKET_COMMONS, MARKETS).collect::<BTreeMap<_, _>>();
+        Self::set_temp_storage(old_markets, "old_markets");
```

Comment:
Isn't this vector a little bit too big? I expected a for loop and to store markets by market id key and value.

Reply:
According to this, there are no size limits for storage values (other than the usual physical limits): https://substrate.stackexchange.com/questions/88/is-there-a-limit-to-the-size-of-a-map-in-pallet-storage. But if this ran on-chain, I'd probably approach this differently.